docs/extending/extending_client_side.md

```diff
@@ -0,0 +1,528 @@
+A Controller is registered with an identifier, a `kebab-case` string that allows the controller to be referenced in data attributes on your HTML, try to keep your Controllers small in scope within the DOM and in their functionality.
+Controllers have methods that define the main behaviour, plus a few built-in callbacks for `connect`, `disconnect`, and others specific to the values set up.
+
+#### [Targets](https://stimulus.hotwired.dev/reference/targets)
+
+Simple Controllers can use the `this.element` to get their controlled element, reach for targets if you need to have access to other DOM elements in your Controller code and avoid using DOM selectors within the JavaScript.
+
+Target elements are attached to the Controller instance with the `data-...-target` attribute, for example `<span data-my-controller-name-target="label"></span>`.
+
+#### [Values](https://stimulus.hotwired.dev/reference/values)
+
+Values are both the initial HTML-driven data for controllers and allow stateful/reactive behaviour. You must declare a value type and an optional default within your controller.
+Values are declared in the initial HTML on the controlled element with the `data-...-value` attribute, for example `<section data-controller="my-controller" data-my-controller-delay-value="300">...</section>`.
+
+#### [Actions](https://stimulus.hotwired.dev/reference/actions)
+
+A simple and powerful way to attach DOM event listeners to any element within your controlled element, letting you react to events such as `click` or `hover` and triggering a Controller method. No JavaScript is needed for this, it's all declarative in your HTML and Stimulus will clean up event listeners for you.
+
+The `data-action` attribute is used, for example `<button type="button" data-action="click->my-controller#open">Open</button>`, where `click` is the event, separated by an `->` and then the controller's identifier with a `#methodName`.
+
+### Adding a custom Stimulus controller
+
+Wagtail exposes a single client-side global (`window.Stimulus`), which is an instance of the core admin Stimulus application.
+
+You need to first create a custom [Stimulus Controller](https://stimulus.hotwired.dev/reference/controllers). This can be done in two main ways, depending on your set-up:
+
+1. Create a **custom Class** that extends the base `window.Stimulus.use.Controller` using class inheritance. If you are using a build tool you can extend `import { Controller } from '@hotwired/stimulus';`.
+2. If you are unable to work with classes, build one with `window.Stimulus.use.createController` which accepts **an object using the [method definitions](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Method_definitions) approach**, with a special `STATIC` property for static properties.
```

Comment:
I’d rather we remove this to be honest. ES6 classes are a good 8 years old and weren’t particularly complex to understand even back then. The more modern `static` syntax isn’t that much harder to understand and has been around for a while too.

Reply:
No problems, I think there may be an edge case here and there that will warrant the usage of this object until. However, it's probably just confusing to have this documented here.

Reply:
Ignoring edge cases, if devs get stuck we can re-add this later or provide some example code for a work around.